warp10/src/main/java/io/warp10/script/MemoryWarpScriptStack.java

```diff
@@ -984,9 +984,10 @@ public void exec(Macro macro) throws WarpScriptException {
       } else {
         String name = macro.getName();
         String section = (String) this.getAttribute(WarpScriptStack.ATTRIBUTE_SECTION_NAME);
-        Object statement = macro.get(i);
         String statementString = "";
-        if (null != statement) {
+        Object statement = null;
+        if (i >= macro.size()) {
+          statement = macro.get(macro.size() - 1);
```

Comment:
The statementString = statement.toString() should be outside of the conditional and statement should be initialized to macro.get(i), not to null.

Reply:
And statementString should be set to String.valueOf(statement)